azurerm/data_source_arm_container_registry.go

```diff
@@ -0,0 +1,109 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Read:          dataSourceArmContainerRegistryRead,
+		MigrateState:  resourceAzureRMContainerRegistryMigrateState,
+		SchemaVersion: 2,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMContainerRegistryName,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+			"location":            locationForDataSourceSchema(),
+			"admin_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"admin_password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"admin_username": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"login_server": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmContainerRegistryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).containerRegistryClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Container Registry %q was not found in Resource Group %q", name, resourceGroup)
+			d.SetId("")
+			return nil
```

Comment:
since this is a Data Source, if this isn't found we should return an error; thus can we make this:
```
return fmt.Errorf("Container Registry %q was not found in Resource Group %q", name, resourceGroup)
```